backend/src/db/entities/lastwill.entity.ts

```diff
@@ -0,0 +1,657 @@
+    The Type is needed for the validation
+    The Transform is needed for the Serialization
+    But the Type also says that it transforms with the subTypes and creates instances, so we are uncertain why this doesn't fix both
+    The Transform is performed after the Type, but in Serialization it seems the Type is not enough transformation
+    This is only needed if lastwillModel returns flattenMaps (for example not on .create)
+  */
+  @Type(() => Discriminator, {
+    discriminator: {
+      property: 'type',
+      subTypes: [
+        // TODO: check if name is the value which is discriminated on, do I have to have all options here?
+        { value: Person, name: PersonType.OTHER },
+        { value: Person, name: PersonType.CHILD },
+        { value: Person, name: PersonType.FATHER },
+        { value: Person, name: PersonType.MOTHER },
+        { value: Person, name: PersonType.SIBLING },
+        { value: Organisation, name: PersonType.ORGANISATION },
+      ],
+    },
+    keepDiscriminatorProperty: true,
+  })
+  @Expose()
+  @Transform(({ value }) =>
+    value?.map((object: Person | Organisation) => {
+      if (object.type === PersonType.ORGANISATION) {
+        return plainToClass(Organisation, object)
+      }
+      return plainToClass(Person, object)
+    }),
+  )
+  heirs: (Person | Organisation)[]
+
+  @prop({ required: true, type: [Item], default: [], _id: false })
+  @ApiProperty({
+    type: [Item],
+    description: 'Items',
+    isArray: true,
+    example: swaggerExampleObject.items,
+  })
+  @ValidateNested({ each: true })
+  @Type(() => Item)
+  @Expose()
+  items: Item[]
+
+  @prop({
+    required: true,
+    type: [FinancialAsset],
+    default: [],
+    _id: false,
+  })
+  @ApiProperty({
+    type: FinancialAsset,
+    description: 'Financial assets',
+    isArray: true,
+    example: swaggerExampleObject.financialAssets,
+  })
+  @ValidateNested({ each: true })
+  @Type(() => FinancialAsset)
+  @Expose()
+  financialAssets: FinancialAsset[]
+
+  @prop({ required: true, default: [], allowMixed: Severity.ALLOW })
+  @ApiProperty({
+    description: 'Progress keys',
+    example: swaggerExampleObject.progressKeys,
+    enum: SidebarPages,
+    type: String,
+    isArray: true,
+  })
+  @IsEnum(SidebarPages, { each: true })
+  @Expose()
+  progressKeys: SidebarPages[]
+
+  constructor(partial: Partial<LastWill>) {
+    Object.assign(this, partial)
+  }
+}
+
+// PickTypes don't work with class-transformer, so we have to create a new class
+export class LastWillMetadata {
+  @ApiProperty({
+    description: 'Progress keys',
+    example: swaggerExampleObject.progressKeys,
+    enum: SidebarPages,
+    type: String,
+    isArray: true,
+  })
+  @Expose()
+  progressKeys: SidebarPages[]
+
+  @ApiProperty({
+    description: 'Testator',
+    example: 'jeff',
+    type: String,
+  })
+  @Expose()
+  testator: string
+
+  constructor(partial: Partial<LastWillMetadata>) {
+    Object.assign(this, partial)
+  }
+}
```

Comment:
I am absolutely disgusted by how much of this type definitions is optional...however there is not much we can do about that

Reply:
I was disgusted as well, makes everything horrible, basically everything is optional, could aswell have no checks at all and say send me whatever u want